File: utio/models/PlotSchedule.js
/**
 * 地块定时执行计划模型
 * 对应表：plot_schedules
 */

class PlotSchedule {
  constructor(data = {}) {
    this.id = data.id || null
    this.plotId = data.plot_id || data.plotId || null
    this.recipeId = data.recipe_id || data.recipeId || null
    this.scheduleTime = data.schedule_time || data.scheduleTime || null
    this.executions = data.executions || 1
    this.isEnabled = data.is_enabled !== undefined ? data.is_enabled : (data.isEnabled !== undefined ? data.isEnabled : 1)
    this.createdAt = data.created_at || data.createdAt || null
    this.updatedAt = data.updated_at || data.updatedAt || null
  }

  /**
   * 转换为数据库字段格式
   */
  toDbFormat() {
    return {
      id: this.id,
      plot_id: this.plotId,
      recipe_id: this.recipeId,
      schedule_time: this.scheduleTime,
      executions: this.executions,
      is_enabled: this.isEnabled ? 1 : 0,
    }
  }

  /**
   * 转换为前端格式
   */
  toFrontendFormat() {
    return {
      id: this.id,
      plotId: this.plotId,
      recipeId: this.recipeId,
      timeHHmm: this.scheduleTime, // 前端使用 timeHHmm 字段名
      executions: parseInt(this.executions),
      isEnabled: this.isEnabled,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt,
    }
  }

  /**
   * 验证数据有效性
   */
  validate() {
    const errors = []
    if (!this.plotId) errors.push('地块ID不能为空')
    if (!this.recipeId) errors.push('配方ID不能为空')
    if (!this.scheduleTime) errors.push('执行时间不能为空')
    if (this.executions < 1) errors.push('执行次数必须大于0')
    // 验证时间格式 HH:mm
    if (this.scheduleTime && !/^([0-1][0-9]|2[0-3]):[0-5][0-9]$/.test(this.scheduleTime)) {
      errors.push('执行时间格式错误，应为 HH:mm')
    }
    return errors
  }
}

module.exports = PlotSchedule

File: backend/src/main/java/com/greenhouse/entity/PlotSchedule.java
package com.greenhouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.util.Date;
import java.time.LocalTime;

/**
 * 地块定时执行计划实体类
 * 对应表：plot_schedules
 */
@Data
@TableName("plot_schedules")
public class PlotSchedule {
    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 地块ID
     */
    private Integer plotId;

    /**
     * 配方ID
     */
    private String recipeId;

    /**
     * 执行时间（HH:mm格式）
     */
    private LocalTime scheduleTime;

    /**
     * 执行周期类型（daily每天/weekly每周/monthly每月）
     * exist = false 表示如果字段不存在则忽略
     */
    @TableField(exist = false)
    private String scheduleType;

    /**
     * 周几（0-6，0=周日，1=周一，...，6=周六）
     * exist = false 表示如果字段不存在则忽略
     */
    @TableField(exist = false)
    private Integer dayOfWeek;

    /**
     * 精确执行时间（年月日时分秒）
     * exist = false 表示如果字段不存在则忽略
     */
    @TableField(exist = false)
    private Date scheduleDatetime;

    /**
     * 上次执行时间
     * exist = false 表示如果字段不存在则忽略
     */
    @TableField(exist = false)
    private Date lastExecutedAt;

    /**
     * 执行次数
     */
    private Integer executions;

    /**
     * 是否启用（0否/1是）
     */
    private Boolean isEnabled;

    /**
     * 创建时间
     */
    private Date createdAt;

    /**
     * 更新时间
     */
    private Date updatedAt;
}
